src/Application/Economy/Commands/BalanceTransactionCommand.cs

```diff
@@ -0,0 +1,31 @@
+using Application.Services;
+using Application.Users.Commands;
+using Domain.Aggregates;
+using Domain.Abstractions;
+using Microsoft.EntityFrameworkCore;
+using FluentValidation;
+using MediatR;
+namespace Application.Economy.Commands;
+
+
+public sealed record BalanceTransactionCommand(User sender, User reciver, decimal amount) : IRequest<bool>;
+
+public sealed class TransactionValidator : AbstractValidator<BalanceTransactionCommand>
+{
+    public TransactionValidator()
+    {
+        RuleFor(x => x.amount).LessThanOrEqualTo(0).WithMessage("amount can not be 0 or less");
+    }
+}
+
+
+
+public sealed class BalanceTransactionCommandHandler(IAppDbContext dbContext, CancellationToken ct)
+    : IRequestHandler<BalanceTransactionCommand, bool>
+{
+    public Task<bool> Handle(BalanceTransactionCommand request,CancellationToken ct)
+    {
+        // request.reciver.Wallet.Balance + request.amount;
```

Comment:
you can use the methods you wrote on the balance here, as well as database update and save operations